src/main/reaktivity/protocol.idl

```diff
@@ -0,0 +1,780 @@
+        {
+            Error error;
+        }
+
+        // 2.8 Definitions
+        enum AmqpRole (AmqpBoolean)
+        {
+            SENDER (0),
+            RECEIVER (1)
+        }
+
+        enum AmqpSenderSettleMode (AmqpUByte)
+        {
+            UNSETTLED (0),
+            SETTLED (1),
+            MIZED (2)
+        }
+
+        enum AmqpReceiverSettleMode (AmqpUByte)
+        {
+            FIRST (0),
+            SECOND (1)
+        }
+
+        typedef AmqpUInt as AmqpHandle;
+        typedef AmqpUInt as AmqpSeconds;
+        typedef AmqpUInt as AmqpMilliseconds;
+        typedef AmqpBinary as AmqpDeliveryTag;
+        typedef AmqpSequenceNo as AmqpDeliveryNumber;
+        typedef AmqpSequenceNo as AmqpTransferNumber;
+        typedef AmqpUInt as AmqpSequenceNo;
+        typedef AmqpUInt as AmqpMessageFormat;
+        typedef AmqpSymbol as AmqpIETFLanguageTag;
+        typedef AmqpMap<AmqpSymbol, V> as AmqpFields<V>;
+
+        list ErrorList (AmqpList)
+        {
+            required AmqpErrorType condition;
+            AmqpString description;
+            AmqpFields info;
+        }
+
+        variant AmqpError switch (AmqpDescribedType+)
+        {
+            case ERROR: ErrorList;
+        }
+
+        enum AmqpErrorType (AmqpSymbol)
+        {
+            INTERNAL_ERROR ("amqp:internal-error"),
+            NOT_FOUND ("amqp:not-found"),
+            UNAUTHORIZED_ACCESS ("amqp:unauthorized-access"),
+            DECODE_ERROR ("amqp:decode-error"),
+            RESOURCE_LIMIT_EXCEEDED ("amqp:resource-limit-exceeded"),
+            NOT_ALLOWED ("amqp:not-allowed"),
+            INVALID_FIELD ("amqp:invalid-field"),
+            NOT_IMPLEMENTED ("amqp:not-implemented"),
+            RESOURCE_LOCKED ("amqp:resource-locked"),
+            PRECONDITION_FAILED ("amqp:precondition-failed"),
+            RESOURCE_DELETED ("amqp:resource-deleted"),
+            ILLEGAL_STATE ("amqp:illegal-state"),
+            FRAME_SIZE_TOO_SMALL ("amqp:frame-size-too-small"),
+            CONNECTION_FORCED ("amqp:connection:forced"),
+            CONNECTION_FRAMING_ERROR ("amqp:connection:framing-error"),
+            CONNECTION_REDIRECT ("amqp:connection:redirect"),
+            SESSION_WINDOW_VIOLATION ("amqp:session:window-violation"),
+            SESSION_ERRANT_LINK ("amqp:session:errant-link"),
+            SESSION_HANDLE_IN_USE ("amqp:session:handle-in-use"),
+            SESSION_UNATTACHED_HANDLE ("amqp:session:unattached-handle"),
+            LINK_DETACH_FORCED ("amqp:link:detach-forced"),
+            LINK_TRANSFER_LIMIT_EXCEEDED ("amqp:link:transfer-limit-exceeded"),
+            LINK_MESSAGE_SIZE_EXCEEDED ("amqp:link:message-size-exceeded"),
+            LINK_REDIRECT ("amqp:link:redirect"),
+            LINK_STOLEN ("amqp:link:stolen"),
+            TRANSACTION_UNKNOWN_ID ("amqp:transaction:unknown-id"),
+            TRANSACTION_ROLLBACK ("amqp:transaction:rollback"),
+            TRANSACTION_TIMEOUT ("amqp:transaction:timeout")
```

Comment:
Suggest putting a newline between groups of errors for increased readability only.